### PR TITLE
Refactor path adaptation

### LINKED ISSUE #311: Refactor path adaptation
In many places we do the same thing:
```ts
const somePath = ...;
const adaptedPath = path.normalize(path.join(config.paths.root, somePath));
```

This could be extracted to an internal function and also it could the do tilde expansion (which we already replace with `process.env.HOME` in `getAccountPath` in `utils.ts`).

We should document the fact that all paths need to be absolute or will be resolved with respect to project root.

### PR DESCRIPTION
**Pull request by Nathan-SL**
## Usage related changes

-   Closes #311

## Development related changes

-   NA

## Checklist:

-   [x] Formatted the code
-   [x] No linter errors + tried to avoid introducing linter warnings
-   [x] Performed a self-review of the code
-   [x] Rebased to the last commit of the target branch (or merged it into my branch)
-   [x] Documented the changes
-   [x] Updated the `test` directory (with a test case consisting of `network.json`, `hardhat.config.ts`, `check.ts`)
-   [x] Linked issues which this PR resolves
-   [x] All tests are passing (for external contributors who don't have access to the CI/CD pipeline)

### CODE REVIEW

File: www/docs/intro.md [hunk 1]
```diff
@@ -517,6 +517,8 @@ module.exports = {
 
 ### Paths
 
+Provide absolute paths when using the plugin. If a relative path is provided, it will be resolved relative to the root of the project.
```
Comment (FabijanC) [inline www/docs/intro.md:520]: Maybe `Prefer providing absolute` instead of `Provide absolute` ?

File: src/utils.ts [hunk 2]
```diff
@@ -136,8 +136,20 @@ export function getArtifactPath(sourcePath: string, paths: ProjectPathsConfig):
     return path.join(paths.starknetArtifacts, suffix);
 }
 
+/**
+ * Adapts path relative to the root of the project
```
Comment (FabijanC) [inline src/utils.ts:140]: Mention that tilde is resolved to homedir

**Review by FabijanC — CHANGES_REQUESTED** [review]
Nice improvement!

File: src/utils.ts (hunk 2 above)
Comment (Nathan-SL) [reply src/utils.ts]: https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/317/commits/f8edb2c0f52589ea5e309a3e67e6b6a289fb6a5c

File: www/docs/intro.md (hunk 1 above)
Comment (Nathan-SL) [reply www/docs/intro.md]: https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/317/commits/f8edb2c0f52589ea5e309a3e67e6b6a289fb6a5c

**Review by FabijanC — APPROVED** [review]

**Pull request merged**